LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/DevAddr.cs

```diff
@@ -27,6 +27,15 @@ namespace LoRaWan
 
         public DevAddr(uint value) => this.value = value;
 
+        public DevAddr(int networkId, int networkAddress)
+            : this((networkId, networkAddress) switch
+            {
+                ( > 0x7F, _) => throw new ArgumentException(null, nameof(networkId)),
+                (_, > (int)NetworkAddressMask) => throw new ArgumentException(null, nameof(networkAddress)),
```

Comment:
Something looks wrong here. The value is being directly compared for equality with the mask. Did you not mean a range check?

Reply:
You're right, I should also take into account negative values (was still thinking in terms of `uint` here, which is wrong). The larger than check on the `NewtorkAddressMask` should be correct, though. Do you think it's wrong?

Reply:
The upper limit check is fine now but also needs to account for negative values.

Reply:
Never mind. For some reason, I was seeing an outdated version of the file.